app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt

```diff
@@ -590,10 +590,10 @@ open class FileListFragment : MultiSelectFragment(
         activitiesRefreshTimer.cancel()
         isLoadingActivities = true
         mainViewModel.currentFolder.value?.let { localCurrentFolder ->
-            FileController.getFileById(localCurrentFolder.id, userDrive)?.let { updatedFolder ->
-                downloadFolderActivities(updatedFolder)
+            mainViewModel.getFileWithDetailsById(localCurrentFolder.id, userDrive) { updatedFolder ->
+                updatedFolder?.let { downloadFolderActivities(updatedFolder) }
                 activitiesRefreshTimer.start()
-            } ?: run { activitiesRefreshTimer.start() }
+            }
```

Comment:
You can avoid callbacks by making the viewmodel method suspend and using the lifecycle of the fragment